core/frontend/src/DisplayStyleState.ts

```diff
@@ -630,6 +670,21 @@ export abstract class DisplayStyleState extends ElementState implements DisplayS
     return this._backgroundMapGeometry.geometry;
   }
 
+  /** @internal */
+  public getGlobalGeometryAndHeightRange(): { geometry: BackgroundMapGeometry, heightRange: Range1d } | undefined {
+    let geometry = this.getIsBackgroundMapVisible() ? this.getBackgroundMapGeometry() : undefined;
+    const terrainRange = ApproximateTerrainHeights.instance.globalHeightRange;
+    let heightRange = this.displayTerrain ? terrainRange : Range1d.createXX(-1, 1);
+    if (this.globeMode === GlobeMode.Ellipsoid && this._contextRealityModels.find((model) => model.isGlobal)) {
+      if (!geometry)
+        geometry = new BackgroundMapGeometry(0, GlobeMode.Ellipsoid, this.iModel);
```

Comment:
Does this get recomputed every time this function is called, if map is off? Could it be computed once and recorded somewhere?

Reply:
I moved the creation to a private property - computed once only when needed.